buildpacks/nodejs-function-invoker/tests/integration_test.rs

```diff
@@ -89,6 +106,59 @@ fn test_function_with_implicit_runtime_dependency_ts() {
     );
 }
 
+fn invoke_function(socket_addr: &SocketAddr, payload: &serde_json::Value) -> serde_json::Value {
+    let id = format!("MyFunction-{}", random_hex_string(10));
+
+    let sf_context = base64_encode_json(&serde_json::json!({
+        "apiVersion": "",
+        "payloadVersion": "",
+        "userContext": {
+           "orgId": "",
+           "userId": "",
+           "username": "",
+           "orgDomainUrl": "",
+           "onBehalfOfUserId": serde_json::Value::Null,
+           "salesforceBaseUrl": ""
+         }
+    }));
+
+    let ssfn_context = base64_encode_json(&serde_json::json!({
+        "resource": "",
+        "requestId": "",
+        "accessToken": "",
+        "apexClassId": serde_json::Value::Null,
+        "apexClassFQN": serde_json::Value::Null,
+        "functionName": "",
+        "functionInvocationId": serde_json::Value::Null
+    }));
+
+    let mut json_body = None;
+
+    retry(
```

Comment:
A bit of a tangent down the internals of the node buildpack internals. I think that we could refactor this `retry` function to return a result and clean up this code:

```rust
pub fn muhretry<T, E>(
    attempts: u32,
    retry_delay_in_secs: u64,
    retryable_action: impl Fn() -> Result<T, E>,
) -> Result<T, E> {
    let mut result = retryable_action();
    for _ in 1..attempts {
        if result.is_ok() {
            return result;
        }
        std::thread::sleep(Duration::from_secs(retry_delay_in_secs));
        result = retryable_action();
    }
    result
}
```

Then here it could be:

```rust
    muhretry(DEFAULT_RETRIES, DEFAULT_RETRY_DELAY_IN_SECONDS, || {
        ureq::post(&format!("http://{socket_addr}"))
            .set("Content-Type", "application/json")
            .set("Authorization", "")
            .set("ce-id", &id)
            .set("ce-time", "2020-09-03T20:56:28.297915Z")
            .set("ce-type", "")
            .set("ce-source", "")
            .set("ce-specversion", "1.0")
            .set("ce-sfcontext", &sf_context)
            .set("ce-sffncontext", &ssfn_context)
            .send_json(payload.clone())
    })
    .map(|res| res.into_json().expect("expected response to be json"))
    .expect("request to assert function health check response failed.")
```

Without needing the intermediate `mut json_body = None`. As a bonus this `muhretry` example is shorter than the current `retry`. The only difference is that it guarantees it will call the code at least once, but I think that's a positive thing.

Looks like it was just added in #663 so I think it's a good time to update.

Reply:
👍 i had considered that but it was already working well in its current form in other places so i left it alone and modified the test instead.  will revisit this decision.

Reply:
Thanks. It wasn't a blocker, but I like this format better.